deepomatic/api/utils.py

```diff
@@ -22,31 +22,38 @@
 THE SOFTWARE.
 """
 
-from deepomatic.api.exceptions import DeepomaticException
-from deepomatic.api.resources.task import Task
-from deepomatic.api.inputs import format_inputs
+import logging
+from tenacity import (Retrying, wait_random_exponential,
+                      stop_after_delay, retry_if_result,
+                      retry_if_exception_type, stop_never,
+                      before_log, after_log)
 
 
-###############################################################################
+logger = logging.getLogger(__name__)
 
-class InferenceResource(object):
-    def inference(self, return_task=False, wait_task=True, **kwargs):
-        assert(self._pk is not None)
 
-        inputs = kwargs.pop('inputs', None)
-        if inputs is None:
-            raise DeepomaticException("Missing keyword argument: inputs")
-        content_type, data, files = format_inputs(inputs, kwargs)
-        result = self._helper.post(self._uri(pk=self._pk, suffix='/inference'), content_type=content_type, data=data, files=files)
-        task_id = result['task_id']
-        task = Task(self._helper, pk=task_id)
-        if wait_task:
-            task.wait()
+class Functor(object):
```

Comment:
I think you could replace this with the standard lib `functools.partial`: cf https://docs.python.org/2/library/functools.html#functools.partial

It is called exactly the same way as your Functor, so for example you could do:

```python
func = functools.partial(self._refresh_tasks_status, pending_tasks, success_tasks, error_tasks, positions)
retry(func,
          retry_if_result(has_pending_tasks),
          **retry_kwargs)
```

Reply:
For some reason I thought It wouldn't work in my case, but now I cannot see the reason why. I will try :)

Reply:
It is working, thanks !